src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -565,7 +571,7 @@ private function validateAndDenormalize(array $types, string $currentClass, stri
                 // PHP's json_decode automatically converts Numbers without a decimal part to integers.
                 // To circumvent this behavior, integers are converted to floats when denormalizing JSON based formats and when
                 // a float is expected.
-                if (Type::BUILTIN_TYPE_FLOAT === $builtinType && \is_int($data) && null !== $format && str_contains($format, JsonEncoder::FORMAT)) {
+                if (Type::BUILTIN_TYPE_FLOAT === $builtinType && \is_int($data) && null !== $format && str_contains($format, static::JSON_ENCODER_FORMAT)) {
```

Comment:
Now that we are allowing userland-defined json format, I wonder how the format match will behave.
I guess that the `str_contains` was before used to handle `json`, `jsonld`, `jsonapi`, ... formats (but I let @dunglas correct me if I'm wrong).

Then, why not simply allow an array of json formats instead?
The default value could contain every basic json formats, and users could be able to add any wanted additional format.
But I think we'll still be stuck with the `str_contains` method for the sake of BC promise.